### PR TITLE
SAK-43901: add wicket library for reusable components

### PR DESCRIPTION
**Pull request by plukasew**
https://jira.sakaiproject.org/browse/SAK-43901

Sakai has several tools that use the Wicket framework. It would be beneficial to have a library of reusable Wicket components for use in these tools, like we have for JSF and Velocity.

This PR creates a Wicket library containing a single component, SakaiDateTimeField, as a starting point. There are two versions of the library, one for tools using Wicket 6, and another for tools using Wicket 8.

SakaiDateTimeField was previously developed for SiteStats (Wicket 6). The PR introduces it to Gradebook (Wicket 6) and Term Manager (Wicket 8). It will also be useful for the SCORM (Wicket 8) contrib tool, which currently has its own copy of SakaiDateTimeField.

Using SakaiDateTimeField in these additional tools also resolves SAK-43658 and SAK-41997.

### CODE REVIEW

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanelContent.html [hunk 1]
```diff
@@ -49,46 +49,8 @@
...
-                                    $datepicker.val("");
-				    // Clear the hidden date too, possibly a better way to do this
-	    			    $datepicker.siblings("#duedate_iso8601").val("")
-                                });
-                            });
-                        };
-                        initDatePickers();
-                    </script>
+				<div class="col-xs-6 form-inline">
+					<input wicket:id="duedate" type="text" class="form-control gb-datepicker" aria-describedby="duedate-help" />
```
[the remark this answers is not in the document]
Comment (plukasew) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanelContent.html:53]: This is a false positive, there is <label wicket:for="duedate" class="col-xs-3 control-label"> a few lines above this.

**Comment by bjones86** [comment]
> There are accessibility issues in these changes.

This is a false positive.

**Comment by ern** [comment]
Can we improve the directory structure to the following:
wicket/pom.xml 
   <artifactId>wicket-tool-base</artifactId>
wicket/wicket6/pom.xml
   <artifactId>wicket-tool-6</artifactId>
wicket/wicket8/pom.xml
   <artifactId>wicket-tool-8</artifactId>

**Comment by plukasew** [comment]
Restructured as requested.

**Comment by steveswinsburg** [comment]
FYI this sort of library used to exist back in the SVN days, but eventually got out of date and then was a problem for everything to use. So a library like this needs to be maintained well.

**Comment by jonespm** [comment]
@steveswinsburg Yeah, I think I found that library and tried it initially, but it was too out of date. Having components is great because you can re-use them but they still need to be kept up-to-date.

**Pull request merged**